relayer/src/tx/connection.rs

```diff
@@ -0,0 +1,121 @@
+use crate::chain::{Chain, CosmosSDKChain};
+use crate::config::ChainConfig;
+use crate::error::{Error, Kind};
+use ibc::ics24_host::identifier::{ClientId, ConnectionId};
+use ibc_proto::connection::{Counterparty, MsgConnectionOpenInit};
+use ibc_proto::tx::v1beta1::TxBody;
+use std::str::FromStr;
+use tendermint::account::Id;
+
+#[derive(Clone, Debug)]
+pub struct ConnectionOpenInitOptions {
+    pub src_client_id: ClientId,
+    pub dest_client_id: ClientId,
+    pub src_connection_id: ConnectionId,
+    pub dest_connection_id: ConnectionId,
+    pub src_chain_config: ChainConfig,
+    pub dest_chain_config: ChainConfig,
+}
+
+pub fn conn_init(opts: ConnectionOpenInitOptions) -> Result<(), Error> {
+    // Get the destination chain
+    let dest_chain = CosmosSDKChain::from_config(opts.clone().dest_chain_config)?;
+
+    let signer = Id::from_str(dest_chain.config().account_prefix.as_str()).map_err(|e| {
+        Kind::MessageTransaction("Connection Open Init: Bad Signer".into()).context(e)
+    })?;
```

Comment:
Something is wrong with the signer and the way it's serialized, when I run the command i get:
```
$ rrly -h -c simple_config.toml tx raw conn-init ibc1 ibc0 ibczeroclient ibconeclient ibczeroconnection ibconeconnection
    Finished dev [unoptimized + debuginfo] target(s) in 0.14s
     Running `target/debug/relayer -h -c simple_config.toml tx raw conn-init ibc1 ibc0 ibczeroclient ibconeclient ibczeroconnection ibconeconnection`
     Message ConnectionOpenInitOptions { src_client_id: ClientId("ibczeroclient"), dest_client_id: ClientId("ibconeclient"), src_connection_id: ConnectionId("ibczeroconnection"), dest_connection_id: ConnectionId("ibconeconnection"), src_chain_config: ChainConfig { id: chain::Id(ibc1), rpc_addr: Tcp { peer_id: None, host: "localhost", port: 26657 }, account_prefix: "cosmos", key_name: "testkey", store_prefix: "ibc", client_ids: ["ibconeclient"], gas: 200000, trusting_period: 1209600s }, dest_chain_config: ChainConfig { id: chain::Id(ibc0), rpc_addr: Tcp { peer_id: None, host: "localhost", port: 26657 }, account_prefix: "cosmos", key_name: "testkey", store_prefix: "ibc", client_ids: ["ibczeroclient"], gas: 200000, trusting_period: 1209600s } }
conn init failed, error:  tx error: Message transaction failure: Connection Open Init: Bad Signer: parse error
```

Reply:
I copied the logic from the client create and noticed it was just using an account_prefix which is usually "cosmos". We need to use an address from the chain. I just hard-coded it for now in the conn-init until we have a way to retrieve that or get it from a context. The command doesn't spit that error anymore using the last commit version.